delocate/libsana.py

```diff
@@ -104,6 +113,63 @@ def resolve_rpath(lib_path, rpaths):
     return lib_path
 
 
+def search_environment_for_lib(lib_path):
+    """ Search common environment variables for `lib_path`
+
+    We'll use a single approach here:
+
+        1. Search for the basename of the library on DYLD_LIBRARY_PATH
+        2. Search for ``realpath(lib_path)``
+        3. Search for the basename of the library on DYLD_FALLBACK_LIBRARY_PATH
+
+    This follows the order that Apple defines for "searching for a
+    library that has a directory name in it" as defined in their
+    documentation here:
+
+    https://developer.apple.com/library/archive/documentation/DeveloperTools/Conceptual/DynamicLibraries/100-Articles/DynamicLibraryUsageGuidelines.html#//apple_ref/doc/uid/TP40001928-SW10
+
+    See the script "testing_osx_rpath_env_variables.sh" in tests/data
+    for a more in-depth explanation. The case where LD_LIBRARY_PATH is
+    used is a narrow subset of that, so we'll ignore it here to keep
+    things simple.
+
+    Parameters
+    ----------
+    lib_path : str
+        Name of the library to search for
+
+    Returns
+    -------
+    lib_path : str
+        Full path of ``basename(lib_path)``'s location, if it can be found, or
+        ``realpath(lib_path)`` if it cannot.
+    """
+    lib_basename = basename(lib_path)
+    potential_library_locations = []
+
+    def _paths_from_var(varname, lib_basename):
```

Comment:
Can't this be at the top level?  It's not using any variables from the enclosing function,  is it?

Reply:
Sure, it absolutely could be -- I put it in here as it was only really used inside this function, but not a problem to move it out.